libs/bsw/middleware/include/middleware/queue/QueueBase.h

```diff
@@ -0,0 +1,230 @@
+        }
+        stats_.realLoadSnapshot += static_cast<uint16_t>(currentSize);
+        ++stats_.realProcessingCounter;
+        if (stats_.previousSnapshot == 0U)
+        {
+            stats_.maxFillRate      = static_cast<uint8_t>(currentSize);
+            stats_.previousSnapshot = static_cast<uint8_t>(currentSize);
+        }
+        else
+        {
+            if (currentSize > stats_.previousSnapshot)
+            {
+                auto const diff = (currentSize - stats_.previousSnapshot);
+                if (diff > stats_.maxFillRate)
+                {
+                    stats_.maxFillRate = static_cast<uint8_t>(diff);
+                }
+            }
+            stats_.previousSnapshot = static_cast<uint8_t>(currentSize);
+        }
+    }
+
+protected:
+    QueueBase() {}
+
+    /**
+     * \brief Init method which needs to be called before doing any work with the queue.
+     *
+     *
+     * \param maxSize the maximum number of elements inside the queue.
+     */
+    void init(uint32_t const maxSize)
+    {
+        maxSize_                     = maxSize;
+        sent_                        = 0U;
+        received_                    = 0U;
+        stats_.processedMessages     = 0U;
+        stats_.lostMessages          = 0U;
+        stats_.loadSnapshot          = 0U;
+        stats_.processingCounter     = 0U;
+        stats_.realLoadSnapshot      = 0U;
+        stats_.realProcessingCounter = 0U;
+        stats_.maxLoad               = 0U;
+        stats_.startupLoad           = 0U;
+        stats_.previousSnapshot      = 0U;
+        stats_.maxFillRate           = 0U;
+    }
+
+    /**
+     * \brief Get the value of received_ attribute.
+     *
+     * \return uint32_t the value of the reading cursor.
+     */
+    uint32_t getReceived() const { return received_; }
+
+    /**
+     * \brief Get the value of sent_ attribute
+     *
+     * \return uint32_t the value of the writing cursor
+     */
+    uint32_t getSent() const { return sent_; }
+
+    /**
+     * \brief Advance the reading cursor.
+     *
+     */
+    void advanceReceived()
+    {
+        received_ = (received_ + 1U) % (2U * maxSize_);
+        ++stats_.processedMessages;
+    }
+
+    /**
+     * \brief Updates the writing cursor to the next element in the buffer.
+     *
+     * \return etl::optional<size_t>
+     */
+    etl::optional<size_t> writeNext()
+    {
+        etl::optional<size_t> writableIndex{};
+        if (isFull())
+        {
+            ++stats_.lostMessages;
+        }
+        else
+        {
+            writableIndex.emplace(sent_ % maxSize_);
+            sent_ = (sent_ + 1U) % (2U * maxSize_);
+            if (size() > stats_.maxLoad)
+            {
+                stats_.maxLoad = static_cast<uint8_t>(size());
+            }
+            if (0U == stats_.processedMessages)
+            {
+                ++stats_.startupLoad;
+            }
+        }
+
+        return writableIndex;
+    }
+
+private:
```

Comment:
According to our coding guidelines the convention for members is an `_underscorePrefix` not `suffix_`.